src/coreclr/vm/debugdebugger.cpp

```diff
@@ -770,242 +943,34 @@ typedef Wrapper<OBJECTHANDLE, DoNothing<OBJECTHANDLE>, HolderDestroyStrongHandle
 // receives a custom notification object from the target and sends it to the RS via
 // code:Debugger::SendCustomDebuggerNotification
 // Argument: dataUNSAFE - a pointer the custom notification object being sent
-FCIMPL1(void, DebugDebugger::CustomNotification, Object * dataUNSAFE)
+extern "C" void QCALLTYPE DebugDebugger_CustomNotification(QCall::ObjectHandleOnStack data)
 {
-    CONTRACTL
-    {
-        FCALL_CHECK;
-    }
-    CONTRACTL_END;
-
-    OBJECTREF pData = ObjectToOBJECTREF(dataUNSAFE);
+    QCALL_CONTRACT;
 
 #ifdef DEBUGGING_SUPPORTED
     // Send notification only if the debugger is attached
-    if (CORDebuggerAttached() )
-    {
-        HELPER_METHOD_FRAME_BEGIN_PROTECT(pData);
+    _ASSERTE(CORDebuggerAttached());
```

Comment:
Nit: There is a small chance that the debugger may detach between the check on the managed side and here.

Reply:
I will re-enable the check.